saleor/graphql/discount/mutations/voucher/voucher_update.py

```diff
@@ -85,12 +91,69 @@ def clean_voucher_usage_setting(cls, instance, data):
                     }
                 )
 
+    @classmethod
+    def count_voucher_usage(cls, voucher: Voucher) -> dict[str, int]:
+        voucher_codes = [code.code for code in voucher.codes.all()]
+        orders = order_models.Order.objects.select_related("channel").filter(
```

Comment:
Why do you need to take this value from order/checkout? In the case of vouchers used in multiple orders/checkouts this query can be super heavy on DB, and also from the memory usage perspective. Why can't we use `VoucherCode.used`?

Reply:
@korycins `code.used` is updated only when the voucher has `usage_limit` set. Here is the case where the limit is set. 
But you are right, this query looks very heavy. Maybe we should consider blocking such behavior when the voucher has already been used.

Reply:
We decided to disallow to change the setting if the voucher has been already used.